frontend/ui/widget/buttondialog.lua

```diff
@@ -90,9 +90,10 @@ function ButtonDialog:init()
         self.width = math.floor(math.min(Screen:getWidth(), Screen:getHeight()) * self.width_factor)
     end
     if self.dismissable then
-        if Device:hasKeys() then
-            local close_keys = Device:hasFewKeys() and { "Back", "Left" } or Device.input.group.Back
-            self.key_events.Close = { { close_keys } }
+        if Device:hasFewKeys() then
+            self.key_events.Close = { { { "Back", "Left", "Menu" } } }
+        elseif not Device:isTouchDevice() then
+            self.key_events.Close = { { { "Back", "Menu" } } }
```

Comment:
But you killed the ability to close it with Backspace or Esc on the emulator.

Reply:
blimey! is that inside the `Device.input.group.Back`?

Reply:
would the `not isTouch` be problematic?

Reply:
Last commit seems to work.

> would the `not isTouch` be problematic?

I don't know

Reply:
I was referring to when you set the emu to be non touch... is that something you do?

Reply:
While I think @poire-z isn't quite right (at present Esc simply sends Back, but while it's not currently a part of that input group it probably should be as a separate entity), the logic nevertheless doesn't seem quite right to me. If we regard this menu thing as desirable we should just change it, and if not I'm not convinced we should add it selectively.

I would make it something like this:
```lua
if hasKeys -- NB! Documentation + not adding unnecessary things to memory.
  local back_group = util.tableDeepCopy(Device.input.group.Back)
  if hasFewKeys -- with left added
    table.insert(back_group, "Left")
    self.key_events.Close = { { { back_group } } }
  else -- regular
    table.insert(back_group, "Menu")
```